tests/integration/high_availability/test_replication.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
```

Comment:
Writes get cleared wtith ` await application_unit.run_action("clear-continuous-writes")` in these tests, but if commands before this call fail, writes to the db will not be cleared. This is a problem if a user provides a pre-exisiting cluster (ie the provided cluster will become full of unwanted writes)

I suggest insteadd using a fixture, the fixture should clear writes regardless of what happens in your tests. You can see this [here](https://github.com/canonical/mongodb-operator/blob/de49e3c11c99f65d5f94204bfa3b0db5606a7f24/tests/integration/ha_tests/test_ha.py#L53)

You might want to adopt fixtures for other changes that you make to a provided cluster since we don't want to leave a lasting effect on a provided cluster

Reply:
addressed in ef4d0f84c9105de5cec89b0c2468f1513e867df3